src/repl.rs

```diff
@@ -234,6 +234,7 @@ where
 
         let readline = rl.readline(prompt);
         match readline {
+            Ok(line) if line.is_empty() => (),
```

Comment:
Oo I think this changes behavior if the `buf` is non-empty and the REPL is in multi-line editing mode.

Does this look better to you?

```suggestion
            Ok(line) if line.is_empty() && buf.is_empty() => (),
```

One way to test this is to try to put this into the REPL and see if the newline is preserved:

```ruby
def foo(x)
  return 1 if x.nil?

  2
end
```

Reply:
OK, so that gets to another issue -- multiline is not behaving at all for me on `trunk`. I'd actually assumed it was unimplemented. 

```sh
~/d/artichoke (trunk)> target/debug/airb
artichoke 0.1.0-pre.0 (2021-08-24 revision 4673) [x86_64-apple-darwin]
[rustc 1.54.0 (a178d0322 2021-07-26) on x86_64-apple-darwin]
>>> def foo(x)
A fatal parsing error occurred
```

Reply:
Yikes. This was never fully working but it looks like it has regressed. My guess is that broke with the mruby 3 upgrade.

Please make the change to only skip blank lines when buf is empty. We can make a ticket for the broken multiline input mode to fix separately.

@stuarth Would you be able to cut the issue for the broken multiline input mode?